datastream-jdbc-connector/src/main/java/com/linkedin/datastream/connectors/jdbc/translator/JdbcCommon.java

```diff
@@ -641,6 +632,15 @@ public static Schema createSchema(final ResultSet rs, AvroConversionOptions opti
                                     : u.stringType());
                     break;
 
+                case TIMESTAMP:
+                case TIMESTAMP_WITH_TIMEZONE:
+                case microsoft.sql.Types.SMALLDATETIME: // -150 represents TSQL smalldatetime type
+                case microsoft.sql.Types.DATETIME: // -151 represents TSQL datetime type
+                case microsoft.sql.Types.DATETIMEOFFSET: // TSQL DATETIMEOFFSET
+                    final Schema timestampMilliType = LogicalTypes.timestampMillis().addToSchema(SchemaBuilder.builder().longType());
```

Comment:
Add a new prop to the schema. Call it "source.sql.type" and set it to datetime, datetimeoffset...

https://avro.apache.org/docs/1.7.2/api/java/org/apache/avro/Schema.Field.html#addProp(java.lang.String,%20java.lang.String)